drivers/interrupt_controller/intc_plic.c

```diff
@@ -21,24 +21,39 @@
 #include <zephyr/drivers/interrupt_controller/riscv_plic.h>
 #include <zephyr/irq.h>
 
-#define PLIC_MAX_PRIO	DT_INST_PROP(0, riscv_max_priority)
-#define PLIC_PRIO	DT_INST_REG_ADDR_BY_NAME_U64(0, prio)
-#define PLIC_IRQ_EN	DT_INST_REG_ADDR_BY_NAME_U64(0, irq_en)
-#define PLIC_REG	DT_INST_REG_ADDR_BY_NAME_U64(0, reg)
-
-#define PLIC_IRQS        (CONFIG_NUM_IRQS - CONFIG_2ND_LVL_ISR_TBL_OFFSET)
-#define PLIC_EN_SIZE     ((PLIC_IRQS >> 5) + 1)
-
-#define PLIC_EDGE_TRIG_TYPE   (PLIC_MAX_PRIO + DT_INST_PROP(0, riscv_trigger_reg_offset))
 #define PLIC_EDGE_TRIG_SHIFT  5
 
 struct plic_regs_t {
 	uint32_t threshold_prio;
 	uint32_t claim_complete;
 };
 
+struct plic_config {
+	mem_addr_t prio;
+	mem_addr_t irq_en;
+	mem_addr_t reg;
+	mem_addr_t trig;
+	uint32_t max_prio;
+	uint32_t num_irqs;
+	riscv_plic_irq_config_func_t irq_config_func;
+};
+
 static int save_irq;
 
+/**
+ * @brief Determine the PLIC device from the IRQ
+ *
+ * FIXME: This function is currently hardcoded to return the first instance.
```

Comment:
right, but this is also making the whole PR basically useless because you cannot match the IRQ number to the actual instance. I don't think it makes sense to merge this without having a fix to this.

Reply:
Yes you are right, this is basically a refactor/cleanup, I was hoping that we can focus on getting the `get_plic_dev_from_irq` right in the other PRs

Reply:
I see but the problem is that we are risking to merge something half-baked without the certainty that the other PRs will be merged and this fixed. Is it too much work to push a fully working PR with all the needed pieces to have this working?